src/Data/List/Fresh/NonEmpty.agda

```diff
@@ -0,0 +1,52 @@
+------------------------------------------------------------------------
+-- The Agda standard library
+--
+-- A non-empty fresh list
+------------------------------------------------------------------------
+
+{-# OPTIONS --without-K --safe #-}
+
+module Data.List.Fresh.NonEmpty where
+
+open import Level using (Level; _⊔_)
+open import Data.List.Fresh as List# using (List#; []; cons; fresh)
+open import Data.Maybe.Base using (Maybe; nothing; just)
+open import Data.Nat using (ℕ; suc)
+open import Data.Product using (_×_; _,_)
+open import Data.Unit.Polymorphic.Base using (⊤; tt)
+open import Relation.Binary as B using (Rel)
+
+private
+  variable
+    a r : Level
+    A : Set a
+
+module _ {a} (A : Set a) (R : Rel A r) where
+
+  infixr 5 _∷#⁺_
+  record List#⁺ : Set (a ⊔ r) where
+    constructor _∷#⁺_
+    field
+      head : A
+      tail : List# A R
+      {rel} : fresh A R head tail
+
+  open List#⁺
+
+  uncons : List#⁺ → A × List# A R
+  uncons (x ∷#⁺ xs) = x , xs
+
+  [_] : A → List#⁺
+  [ x ] = record { head = x; tail = []; rel = tt}
```

Comment:
Can you not use the `_∷#⁺_` constructor here? I would think that Agda should be able to infer `tt`?